alert-common/src/main/java/com/synopsys/integration/alert/common/event/EventListenerConfigurer.java

```diff
@@ -29,38 +29,51 @@
 import org.springframework.jms.config.JmsListenerEndpointRegistrar;
 import org.springframework.jms.config.SimpleJmsListenerEndpoint;
 
 @Configuration
 public class EventListenerConfigurer implements JmsListenerConfigurer {
     private final Logger logger = LoggerFactory.getLogger(EventListenerConfigurer.class);
 
-    private final List<AlertEventListener> alertEventListeners;
+    private final List<AlertChannelEventListener> alertChannelEventListeners;
+    private final List<AlertDefaultEventListener> alertDefaultEventListeners;
+    private final DefaultJmsListenerContainerFactory defaultJmsListenerContainerFactory;
+    private final DefaultJmsListenerContainerFactory distributionChannelJmsListenerContainerFactory;
 
     @Autowired
-    public EventListenerConfigurer(final List<AlertEventListener> alertEventListeners) {
-        this.alertEventListeners = alertEventListeners;
+    public EventListenerConfigurer(List<AlertChannelEventListener> alertChannelEventListeners,
+        List<AlertDefaultEventListener> alertDefaultEventListeners,
+        DefaultJmsListenerContainerFactory defaultJmsListenerContainerFactory,
+        DefaultJmsListenerContainerFactory distributionChannelJmsListenerContainerFactory) {
+        this.alertChannelEventListeners = alertChannelEventListeners;
+        this.alertDefaultEventListeners = alertDefaultEventListeners;
+        this.defaultJmsListenerContainerFactory = defaultJmsListenerContainerFactory;
+        this.distributionChannelJmsListenerContainerFactory = distributionChannelJmsListenerContainerFactory;
     }
 
     @Override
-    public void configureJmsListeners(final JmsListenerEndpointRegistrar registrar) {
+    public void configureJmsListeners(JmsListenerEndpointRegistrar registrar) {
         logger.info("Registering JMS Listeners");
-        for (final AlertEventListener listener : alertEventListeners) {
-            if (listener != null) {
-                final String destinationName = listener.getDestinationName();
-                final String listenerId = createListenerId(destinationName);
-                logger.info("Registering JMS Listener: {}", listenerId);
-                final SimpleJmsListenerEndpoint endpoint = new SimpleJmsListenerEndpoint();
-                endpoint.setId(listenerId);
-                endpoint.setDestination(destinationName);
-                endpoint.setMessageListener(listener);
-                registrar.registerEndpoint(endpoint);
-            }
+        alertDefaultEventListeners.forEach(listener -> registerListenerEndpoint(registrar, listener, defaultJmsListenerContainerFactory));
+        alertChannelEventListeners.forEach(listener -> registerListenerEndpoint(registrar, listener, distributionChannelJmsListenerContainerFactory));
+    }
+
+    private void registerListenerEndpoint(JmsListenerEndpointRegistrar registrar, AlertEventListener listener, DefaultJmsListenerContainerFactory jmsListenerContainerFactory) {
+        if (listener != null) {
```

Comment:
Why would this ever be null?

Reply:
This is a copy-paste from the previous implementation in configureJmsListeners. But you may be correct, I can't think of a case where the listener would come back as null

Reply:
Let's figure it out and (hopefully) get rid of the null-check.